private/bufpkg/bufimage/bufimagemodify/bufimagemodify.go

```diff
@@ -101,46 +107,68 @@ func GoPackage(
+	overrides map[string]string,
 ) (Modifier, error) {
 	return goPackage(
 		sweeper,
 		defaultImportPathPrefix,
 		except,
-		override,
+		moduleOverrides,
+		overrides,
 	)
 }
 
 // JavaMultipleFiles returns a Modifier that sets the java_multiple_files
 // file option to the given value in all of the files contained in
 // the Image.
-func JavaMultipleFiles(sweeper Sweeper, value bool) Modifier {
-	return javaMultipleFiles(sweeper, value)
+func JavaMultipleFiles(sweeper Sweeper, value bool, overrides map[string]string) (Modifier, error) {
+	validatedOverrides, err := stringOverridesToBoolOverrides(overrides)
+	if err != nil {
+		return nil, fmt.Errorf("invalid override for %s: %w", JavaMultipleFilesID, err)
+	}
+	return javaMultipleFiles(sweeper, value, validatedOverrides), nil
 }
 
 // JavaOuterClassname returns a Modifier that sets the java_outer_classname file option
 // in all of the files contained in the Image based on the PascalCase of their filename.
-func JavaOuterClassname(sweeper Sweeper) Modifier {
-	return javaOuterClassname(sweeper)
+func JavaOuterClassname(sweeper Sweeper, overrides map[string]string) Modifier {
+	return javaOuterClassname(sweeper, overrides)
 }
 
 // JavaPackage returns a Modifier that sets the java_package file option
 // according to the given packagePrefix.
-func JavaPackage(sweeper Sweeper, packagePrefix string) (Modifier, error) {
-	return javaPackage(sweeper, packagePrefix)
+func JavaPackage(sweeper Sweeper, packagePrefix string, overrides map[string]string) (Modifier, error) {
+	return javaPackage(sweeper, packagePrefix, overrides)
 }
 
 // JavaStringCheckUtf8 returns a Modifier that sets the java_string_check_utf8 file option according
 // to the given value.
-func JavaStringCheckUtf8(sweeper Sweeper, value bool) Modifier {
-	return javaStringCheckUtf8(sweeper, value)
+func JavaStringCheckUtf8(sweeper Sweeper, value bool, overrides map[string]string) (Modifier, error) {
+	validatedOverrides, err := stringOverridesToBoolOverrides(overrides)
+	if err != nil {
+		return nil, fmt.Errorf("invalid override for %s: %w", JavaStringCheckUtf8ID, err)
+	}
+	return javaStringCheckUtf8(sweeper, value, validatedOverrides), nil
 }
 
 // OptimizeFor returns a Modifier that sets the optimize_for file
 // option to the given value in all of the files contained in
 // the Image.
-func OptimizeFor(sweeper Sweeper, value descriptorpb.FileOptions_OptimizeMode) Modifier {
-	return optimizeFor(sweeper, value)
+func OptimizeFor(
+	sweeper Sweeper,
+	value descriptorpb.FileOptions_OptimizeMode,
+	overrides map[string]string,
+) (Modifier, error) {
+	validatedOverrides := map[string]descriptorpb.FileOptions_OptimizeMode{}
+	for fileImportPath, overrideString := range overrides {
+		overrideOptimizeMode, err := overrideStringToFileOptionOptimizeMode(overrideString)
+		if err != nil {
+			return nil, fmt.Errorf("invalid override for modifier %s and file %s: %w", OptimizeForID, fileImportPath, err)
```

Comment:
Even though it's only used in one place, it might be worth making a helper function for the multi-`OptimizeMode` override validation so that it matches the error structure of the others (e.g. `stringOverridesToBoolOverrides`).

That way we can do something like:
```
validatedOverrides, err := stringOverridesToOptimizeModeOverrides(overrides)
if err != nil {
  return nil, fmt.Errorf("invalid override for %s: %w", OptimizeForID, err)
}
```

Otherwise we'll have different error messages for different modifier IDs, which isn't great for the CLI UX.